### PR TITLE
_1password: 1.7.0 -> 1.8.0

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Installed shell completion as this was added a while ago, and reordered everything a bit.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/1password/default.nix [hunk 1]
```diff
@@ -26,13 +26,22 @@ stdenv.mkDerivation rec {
     zcat Payload | cpio -i
   '';
 
+  dontStrip = stdenv.isDarwin;
+
+  postPhases = [ "postInstall" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/1password/default.nix:31]: I am not sure if it is cool overwriting this phase.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107442` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>_1password</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107442` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>_1password</li>
  </ul>
</details>

File: pkgs/applications/misc/1password/default.nix (hunk 1 above)
Comment (IvarWithoutBones) [reply pkgs/applications/misc/1password/default.nix:31]: Without overwriting this, the phase never runs and shell completion never gets installed. I'm not sure there's a better solution.

Comment (SuperSandro2000) [reply pkgs/applications/misc/1password/default.nix:31]: That is because installPhase is overwritten and not calling pre and post hooks.

Comment (IvarWithoutBones) [reply pkgs/applications/misc/1password/default.nix:31]: When you do call those hooks, the binary never gets patched. Not sure how to approach this, diff i used + error are below.
```diff
diff --git a/pkgs/applications/misc/1password/default.nix b/pkgs/applications/misc/1password/default.nix
index 9e52e1ac244..56d8d402e15 100644
--- a/pkgs/applications/misc/1password/default.nix
+++ b/pkgs/applications/misc/1password/default.nix
@@ -28,12 +28,12 @@ stdenv.mkDerivation rec {

   dontStrip = stdenv.isDarwin;

-  postPhases = [ "postInstall" ];
-
   nativeBuildInputs = [ installShellFiles ] ++ stdenv.lib.optionals stdenv.isLinux [ autoPatchelfHook ];

   installPhase = ''
+    runHook preInstall
     install -D op $out/bin/op
+    runHook postInstall
   '';

   # This needs to be done after patchelf has been ran on the binary
```
```shell
building '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv'...
builder for '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv' failed with exit code 127; last 10 log lines:
  unpacking sources
  unpacking source archive /nix/store/cnlnslraf8zlw1c2158bdl7nnkw94gdg-source
  source root is source
  patching sources
  configuring
  no configure script, doing nothing
  building
  no Makefile, doing nothing
  installing
  /nix/store/7d2w3lafm8cla7n6ii5jskhsnqg4zv3a-stdenv-linux/setup: line 81: /nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op: No such file or directory
[0 built (1 failed)]
error: build of '/nix/store/pb0c6xx470ir0bij7b1qyv99f3g2fa5h-1password-1.8.0.drv' failed
~/nix/nixpkgs/pkgs/applications/misc/1password > file /nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op
/nix/store/n0j5i8a7x1y3clibjvyfs0xb5ll2snf5-1password-1.8.0/bin/op: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, Go BuildID=JIw--AYYM5GRNoEakTfx/NT4DSbHHI2jWoJQdw-mg/Wp_rILnTzXvcSq99zeAp/9Dl6huS2F26iSE1Jm78q, stripped

```

File: pkgs/applications/misc/1password/default.nix [hunk 2]
```diff
@@ -41,11 +50,11 @@ stdenv.mkDerivation rec {
   '';
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/1password/default.nix:52]:
```suggestion
  meta = with lib; {
```

File: pkgs/applications/misc/1password/default.nix [hunk 3]
```diff
@@ -26,13 +26,22 @@ stdenv.mkDerivation rec {
     zcat Payload | cpio -i
   '';
 
+  dontStrip = stdenv.isDarwin;
+
+  postPhases = [ "postInstall" ];
+
+  nativeBuildInputs = [ installShellFiles ] ++ stdenv.lib.optionals stdenv.isLinux [ autoPatchelfHook ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/1password/default.nix:33]:
```suggestion
  nativeBuildInputs = [ installShellFiles ]
    ++ lib.optionals stdenv.isLinux [ autoPatchelfHook ];
```

**Pull request closed without merging**